### DISCUSSION TITLE
During Initialization, two plugins "Clone succeeded, but checkout failed."

### DISCUSSION THREAD

**Discussion by GymRat102** [ayamir/nvimdots#863] [Q&A] [answered]
Hi comminity,

As title says, I encountered error during plugin initialization like this:

<img width="905" alt="image" src="https://github.com/ayamir/nvimdots/assets/35257543/d8dc7bb2-4185-419e-9da3-d91048a79e9d">

Retried by re-entering nvim several times but didn't work. The log says `retry with 'git restore --source=HEAD :/'` but I don't know where to execute this command.

Appreciate for help

**Comment by charliie-dev** [comment]
U seem to have a failed installation of the plugins. Just remove the plugin folder and re-open nvim and let the plugin reinstall itself again. 

If u don't know where to find the plugin folder, check our repo wiki.

**Reply by GymRat102** [reply]
The same error happened again after I removed those two folder under `~/.local/share/nvim/site/lazy`. 

Following the log, I executed `git restore --source=HEAD :/` in plugin folder and all repo files appeared:

<img width="461" alt="image" src="https://github.com/ayamir/nvimdots/assets/35257543/9dd82186-d090-44f4-aed6-bb1261093cbb">

But when I re-entered nvim it started to re-install again and same error happened.

**Reply by Jint-lzxy** [reply]
macOS's git doesn't support overly long pathnames by default, you need to enable this (feature) manually:
```console
git config --global core.longpaths true
```

**Reply by GymRat102** [reply]
hi, sorry for late response. The config doesn't work😰, even though I try several times. Still can't install `copilot.lua` and `markdown-preview.nvim` these two plugins. Stuck and timed-out:

<img width="1245" alt="image" src="https://github.com/ayamir/nvimdots/assets/35257543/4ecc8bbf-1cd5-4d0b-aa1b-8f9d2d491362">

<img width="310" alt="image" src="https://github.com/ayamir/nvimdots/assets/35257543/796a28e2-9665-40f7-9aa8-890ce42da2f8">

by the way, I've seen people mention `core.symlinks` config of git, but it doesn't work as well.

This issue might be unique for my machine setup, I am gonna explore more solutions people mentioned and see if they work. Thanks!

**Reply by Jint-lzxy** [reply]
What u see for `git status` and `git restore --source=HEAD :/`?

**Reply by GymRat102** [reply]
this is what it looks like for `~/.local/share/nvim/site/lazy/copilot.lua`:

<img width="451" alt="image" src="https://github.com/ayamir/nvimdots/assets/35257543/11151626-93e0-447a-a362-e26e1f24fc82">

when I run `git status`:

<img width="736" alt="image" src="https://github.com/ayamir/nvimdots/assets/35257543/9dd79b71-c3eb-411f-acb4-bca6d4ac0ac9">


then when I run `git restore --source=HEAD :/`:

<img width="449" alt="image" src="https://github.com/ayamir/nvimdots/assets/35257543/39276609-0ce5-4afc-92a1-874ee5629bd7">

everything appears.

**Reply by Jint-lzxy** [reply]
hmm this is more like the result of (external) intervention, as `git checkout` never fails due to deleted nodes.

Can u provide those relevant information (if possible) after removing the _whole_ plugin directory?
```sh
rm -rf ~/.local/share/nvim/site/lazy && nvim
```

**Reply by GymRat102** [reply]
it first showed this prompt and then entered into plugin installation

<img width="1228" alt="image" src="https://github.com/ayamir/nvimdots/assets/35257543/49b859e8-3606-403a-be4c-7ce1913f23c0">

<img width="1316" alt="image" src="https://github.com/ayamir/nvimdots/assets/35257543/4aa2c879-55ab-4748-a280-718a3f7431d2">

This time more plugin failed. Do you think it could be due to network connectivity? Another mac installed successfully (with a private VPN connect)

**Reply by Jint-lzxy** [reply, marked as answer]
Did u see any difference wrt the output of `git status` and `git restore --source=HEAD :/`? If not, then this is indeed an issue with network connectivity - sometimes ur computer may receive corrupted data packs and this makes such data seems inaccessible. This explains why clone succeeded but checkout failed.

**Reply by GymRat102** [reply]
network connectivity is the cause! I changed to a stable one and it worked. Thank you for your patience @Jint-lzxy @CharlesChiuGit
